src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,162 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final int RESIZE_FACTOR = 2;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_INITIAL_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> newNode = new Node<>(hash(key), key, value, null);
+        int tableIndex = hash(key);
+        if (table[tableIndex] != null) {
+            Node<K, V> temp = table[tableIndex];
+            Node<K, V> nextNode;
+            if (Objects.equals(table[tableIndex].key, key)) {
+                table[tableIndex] = newNode;
+                table[tableIndex].next = temp.next;
+                return;
+            }
+            while (temp.next != null) {
+                if (hash(table[tableIndex].key) == hash(key)) {
+                    if (Objects.equals(temp.next.key, key)) {
+                        nextNode = temp.next;
+                        temp.next = newNode;
+                        newNode.next = nextNode;
+                        return;
+                    }
+                    temp = temp.next;
+                }
+            }
+            temp.next = newNode;
+        } else {
+            table[tableIndex] = newNode;
+        }
+        if (++size > threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        int tableIndex = hash(key);
+        if (table[tableIndex] == null) {
+            return null;
+        }
+        if (table[tableIndex].next != null) {
+            Node<K, V> temp = table[tableIndex];
+            while (temp != null && !Objects.equals(temp.key, key)) {
+                temp = temp.next;
+            }
+            if (temp == null) {
+                return null;
+            }
+            return temp.value;
+        }
+        return table[tableIndex].getValue();
```

Comment:
Don't use getters and setters in class Node. Direct field access is preferred in the context of an inner class.